packages/gatsby/src/utils/merge-gatsby-config.ts

```diff
@@ -46,15 +32,43 @@ const howToMerge = {
    * This makes sure that if a single value is defined, that one it used.
    * We prefer the "right" value, because the user's config will be "on the right"
    */
-  byDefault: (a, b) => b || a,
-  siteMetadata: (objA, objB) => _.merge({}, objA, objB),
+  byDefault: (a: ConfigKey, b: ConfigKey): ConfigKey => b || a,
+  siteMetadata: (objA: Metadata, objB: Metadata): Metadata =>
+    _.merge({}, objA, objB),
   // plugins are concatenated and uniq'd, so we don't get two of the same plugin value
-  plugins: (a = [], b = []) =>
+  plugins: (a: Plugins = [], b: Plugins = []): Plugins =>
```

Comment:
I think it should be `PluginEntry[]` and not `Plugins` (across the board for args and return of this merge function)